frontend/src/components/activity/content/LAThematicArea.vue

```diff
@@ -1,46 +1,204 @@
-      </v-list-item-group>
-    </v-list>
+      </template>
+    </e-select>
   </card-content-node>
 </template>
 
 <script>
-import ApiCheckbox from '@/components/form/api/ApiCheckbox.vue'
 import CardContentNode from '@/components/activity/CardContentNode.vue'
 import { contentNodeMixin } from '@/mixins/contentNodeMixin.js'
+import { debounce, isEqual, sortBy } from 'lodash'
+import { serverErrorToString } from '@/helpers/serverError.js'
 
 export default {
   name: 'LAThematicArea',
-  components: { CardContentNode, ApiCheckbox },
+  components: { CardContentNode },
   mixins: [contentNodeMixin],
+  data() {
+    return {
+      localSelection: [],
+      isSaving: false,
+      savingRequestCount: 0,
+      dirty: false,
+      errorMessages: [],
+      debouncedSave: () => null,
+    }
+  },
+  computed: {
+    dataOptions() {
+      return Object.entries(this.contentNode.data.options)
+    },
+    serverSelection() {
+      return this.dataOptions.filter(([_, option]) => option.checked).map(([key]) => key)
+    },
+    selectionCount() {
+      return this.localSelection.length
+    },
+    items() {
+      return this.dataOptions.map(([key, option]) => ({
+        text: this.$tc(`contentNode.laThematicArea.entity.option.${key}.name`),
+        description: this.$tc(
+          `contentNode.laThematicArea.entity.option.${key}.description`
+        ),
+        value: key,
+        checked: option.checked,
+      }))
+    },
+  },
+  watch: {
+    serverSelection: {
+      async handler(newOptions, oldOptions) {
+        if (isEqual(sortBy(newOptions), sortBy(oldOptions))) {
+          return
+        }
+
+        // copy incoming data if not dirty or if incoming data is the same as local data
+        if (!this.dirty || isEqual(sortBy(newOptions), sortBy(this.localSelection))) {
+          this.resetLocalData()
+        }
+      },
+      immediate: true,
+    },
+  },
+  created() {
+    const DEBOUNCE_WAIT = 500
+    this.debounceSave = debounce(this.save, DEBOUNCE_WAIT)
+  },
   methods: {
-    async refreshContent() {
-      await this.api.reload(this.contentNode)
+    onInput() {
+      this.isSaving = true
+      this.dirty = true
+      this.errorMessages = []
+
+      this.debounceSave()
+    },
+    save() {
+      this.savingRequestCount++
+      this.contentNode
+        .$patch({
+          data: {
+            options: Object.fromEntries(
+              this.dataOptions.map(([key]) => [
+                key,
+                { checked: this.localSelection.includes(key) },
+              ])
+            ),
+          },
+        })
+        .then(() => {
+          this.savingRequestCount--
+          if (this.savingRequestCount === 0) {
+            this.isSaving = false
+          }
+        })
+        .catch((e) => {
+          this.isSaving = false
+          this.errorMessages.push(serverErrorToString(e))
+        })
+    },
+    resetLocalData() {
+      this.localSelection = [...this.serverSelection]
```

Comment:
you should also decrease the savingRequestCount here, else if a previous request fails, the component is stuck in the saving mode.
(or maybe decrease it in the finally for both)

Reply:
Good point!